Geometry/HGCalGeometry/test/HGCalTestRecHitTool.cc

```diff
@@ -0,0 +1,180 @@
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESTransientHandle.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/ForwardDetId/interface/HGCalDetId.h"
+#include "DataFormats/HcalDetId/interface/HcalDetId.h"
+#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+#include "Geometry/HcalCommonData/interface/HcalDDDRecConstants.h"
+#include "Geometry/HcalTowerAlgo/interface/HcalGeometry.h"
+#include "Geometry/HGCalCommonData/interface/HGCalDDDConstants.h"
+#include "Geometry/HGCalGeometry/interface/HGCalGeometry.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
+#include "CoralBase/Exception.h"
+
+class HGCalTestRecHitTool : public edm::one::EDAnalyzer<> {
+public:
+  explicit HGCalTestRecHitTool(const edm::ParameterSet& );
+  ~HGCalTestRecHitTool();
+
+  void beginJob() override {}
+  void analyze(edm::Event const& iEvent, edm::EventSetup const&) override;
+  void endJob() override {}
+  
+private:
+  std::vector<double> retrieveLayerPositions(unsigned int);
+  template<typename GEOM> void check_geom(const GEOM* geom) const;
+  template<typename DDD>  void check_ddd(const DDD* ddd) const;
+  const HGCalDDDConstants* get_ddd(const HGCalDetId& detid) const;
+  const HcalDDDRecConstants* get_ddd(const HcalDetId& detid) const;
+  double      getLayerZ(DetId const&) const;
+  double      getLayerZ(int type, int layer) const;
+  GlobalPoint getPosition(DetId const&) const;
+
+  const CaloGeometry *geom_;
+};
+
+HGCalTestRecHitTool::HGCalTestRecHitTool(const edm::ParameterSet& iC) { }
+
+
+HGCalTestRecHitTool::~HGCalTestRecHitTool() {}
+
+void HGCalTestRecHitTool::analyze(const edm::Event& , 
+				  const edm::EventSetup& iSetup) {
+
+  edm::ESHandle<CaloGeometry> pG;
+  iSetup.get<CaloGeometryRecord>().get(pG);
+
+  if (pG.isValid()) {
+    geom_ = pG.product();
+    retrieveLayerPositions(52);
+  } else {
+    std::cout << "Cannot get valid CaloGeometry Object" << std::endl;
+  }
+}
+
+template<typename GEOM>
+void HGCalTestRecHitTool::check_geom(const GEOM* geom) const {
+  if (nullptr == geom) {
+    throw cms::Exception("HGCalTestRecHitTools")
+      << "Geometry not provided yet to HGCalTestRecHitTools!";
+  }
+}
+
+template<typename DDD>
+void HGCalTestRecHitTool::check_ddd(const DDD* ddd) const {
+  if (nullptr == ddd) {
+    throw cms::Exception("HGCalTestRecHitTools")
+      << "DDDConstants not accessibl to HGCalTestRecHitTools!";
+  }
+}
+
+std::vector<double> HGCalTestRecHitTool::retrieveLayerPositions(unsigned layers) {
+
+  DetId id;
+  std::vector<double> layerPositions;
+  for (unsigned ilayer=1; ilayer<=layers; ++ilayer) {
+    int lay(ilayer), type(0);
+    if (ilayer<=28) {
```

Comment:
@bsunanda we really need a way to get these magic numbers from a geometry/topology class